alyx/actions/views.py

```diff
@@ -90,21 +90,34 @@ def get_queryset(self):
 
 
 class SessionFilter(FilterSet):
-    subject = django_filters.CharFilter('subject__nickname')
-    start_date = django_filters.CharFilter('start_time__date', lookup_expr=('exact'))
-    end_date = django_filters.CharFilter('end_time__date', lookup_expr=('exact'))
-    starts_before = django_filters.CharFilter('start_time__date', lookup_expr=('lte'))
-    starts_after = django_filters.CharFilter('start_time__date', lookup_expr=('gte'))
-    ends_before = django_filters.CharFilter('start_time__date', lookup_expr=('lte'))
-    ends_after = django_filters.CharFilter('start_time__date', lookup_expr=('gte'))
-    dataset_types = django_filters.CharFilter('dataset_types', method='filter_dataset_types')
+    subject = django_filters.CharFilter(name='subject__nickname', lookup_expr=('iexact'))
+    dataset_types = django_filters.CharFilter(name='dataset_types', method='filter_dataset_types')
+    users = django_filters.CharFilter(name='users__username', method=('filter_users'))
+    date_range = django_filters.CharFilter(name='date_range', method=('filter_date_range'))
+    type = django_filters.CharFilter(name='type', lookup_expr=('iexact'))
+
+    def filter_users(self, queryset, name, value):
+        users = value.split(',')
+        queryset = queryset.filter(users__username__in=users)
+        queryset = queryset.annotate(
+            dtypes_count=Count('users__username'))
+        queryset = queryset.filter(dtypes_count__gte=len(users))
+        return queryset
+
+    def filter_date_range(self, queryset, name, value):
+        drange = value.split(',')
+        queryset = queryset.filter(
+            Q(start_time__date__gte=drange[0]) | Q(end_time__date__gte=drange[0]),
+            Q(start_time__date__lte=drange[1]) | Q(end_time__date__lte=drange[1]),
```

Comment:
how does that work when there is only a start or end date?
may be worth adding some tests too